src/main/java/com/google/sps/servlets/AddSchoolData.java

```diff
@@ -152,6 +140,28 @@ private Entity createTerm(String term, Key professor, Key parent) {
     Entity newTerm = new Entity("Term", parent);
     newTerm.setProperty("term", term);
     newTerm.setProperty("professorKey", professor);
+    GraphDataObject bug = new GraphDataObject("bug", "bug");
+    newTerm.setProperty("bug", bug);
     return newTerm;
   }
+
+  private void putNewEntities(Entity school, Entity course, Entity term, Entity professor)
+      throws DatastoreFailureException, IllegalArgumentException {
+    try {
+      if (school != null) {
+        db.put(school);
+      }
+      if (course != null) {
+        db.put(course);
+      }
+      if (term != null) {
+        db.put(term);
+      }
+      if (professor != null) {
+        db.put(professor);
+      }
+    } catch (DatastoreFailureException | IllegalArgumentException e) {
+      throw e;
+    }
```

Comment:
Nice job, another advantage of co-locating the puts is you can do something like this.